packages/arb-bridge-peripherals/scripts/deploy_token_bridge_l1.ts

```diff
@@ -0,0 +1,103 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { concat, id, keccak256, zeroPad } from 'ethers/lib/utils'
+import { ethers } from 'hardhat'
+import deployments from '../deployment.json'
+
+import { Bridge } from 'arb-ts/src'
+import { writeFileSync } from 'fs'
+
+const main = async () => {
+  const accounts = await ethers.getSigners()
+  // TODO: check buddy deployer address available
+  // TODO: check 1820 registry
+  const inboxAddress =
+    process.env.INBOX_ADDRESS || '0x0d0c1aDf6523D422ec7192506A7F6790253399fB'
+
+  if (inboxAddress === '' || inboxAddress === undefined)
+    throw new Error('Please set inbox address! INBOX_ADDRESS')
+
+  console.log('deployer', accounts[0].address)
+
+  const SafeERC20Namer = await ethers.getContractFactory('SafeERC20Namer')
+  const safeERC20Namer = await SafeERC20Namer.deploy()
+
+  const EthERC20Bridge = await ethers.getContractFactory('EthERC20Bridge', {
+    libraries: {
+      SafeERC20Namer: safeERC20Namer.address,
+    },
+  })
+
+  const gasPrice = 0
+  const maxGas = 100000000000
+  const ethERC20Bridge = await EthERC20Bridge.deploy(
+    inboxAddress,
+    deployments.buddyDeployer,
+    maxGas,
+    gasPrice,
+    deployments.standardArbERC777,
+    deployments.standardArbERC20
+  )
+  const arbTokenBridge = await ethERC20Bridge.l2Buddy()
+  console.log('EthERC20Bridge deployed to:', ethERC20Bridge.address)
+  console.log('L2 ArbBridge deployed to:', arbTokenBridge)
+
+  const contracts = JSON.stringify({
+    ...deployments,
+    ethERC20Bridge: ethERC20Bridge.address,
+    arbTokenBridge: arbTokenBridge,
+  })
+  const deployFilePath = './deployment.json'
+  console.log(`Writing to JSON at ${deployFilePath}`)
+  writeFileSync(deployFilePath, contracts)
+
+  const deployReceipt = await ethers.provider.getTransactionReceipt(
+    ethERC20Bridge.deployTransaction.hash
+  )
+
+  const inboxEventSignature = [
+    id('InboxMessageDelivered(uint256,bytes)'),
+    id('InboxMessageDeliveredFromOrigin(uint256)'),
+  ]
+  const inboxEvent = deployReceipt.logs.filter(
+    log =>
+      log.topics[0] === inboxEventSignature[0] ||
+      log.topics[0] === inboxEventSignature[1]
+  )
+
+  if (inboxEvent.length !== 1) {
+    console.log(inboxEvent)
+    throw new Error('Triggered inbox multiple times?')
+  }
+  const inboxSequenceNumber = inboxEvent[0].topics[1]
+
+  const l2DeployTxHash = keccak256(
+    concat([
+      zeroPad(deployments.l2ChainId, 32),
+      zeroPad(inboxSequenceNumber, 32),
+    ])
+  )
```

Comment:
I think all of this stuff would better be replaced by the unique send id system. In the `L2Deployer`, `sendTxToL1` now returns a unique ID which is one of the indexes of the `L2ToL1Transaction` event

The `L2Deployer` `Deployed` event should include the unique id as a field. Then this code can just do an event query for the `Deployed event based on `ethERC20Bridge.address` being the sender and get the unique id from that which can then be used to find the `L2ToL1Transaction` event

That's basically the idea for kicking the problem of discovering relevant L2 to L1 transactions up to the bridge level